crypto/x509/x509_vfy.c

```diff
@@ -324,14 +325,26 @@ static X509 *find_issuer(X509_STORE_CTX *ctx, STACK_OF(X509) *sk, X509 *x)
 static int check_issued(X509_STORE_CTX *ctx, X509 *x, X509 *issuer)
 {
     int ret;
-    if (x == issuer)
-        return cert_self_signed(x);
+    int ss;
+
+    if (x == issuer) {
+        ss = cert_self_signed(ctx, x);
+        if (ss < 0)
+            return X509_V_ERR_UNSPECIFIED;
```

Comment:
This does not look right. You're returning verify error value where the check_issued is expected to be returning only true/false.

Reply:
Fixed.